### PR TITLE
Created Mongoose connection and implemented MetagameEvent handler

### LINKED ISSUE #48: Integrate Database connection
~So we've had some discussion and for now it seems logical to use TypeORM. It saves a lot of headaches creating schemas etc and it allows us to use entities which we just save.~

Changed to Mongoose

I've been assured that it doesn't have any massive performance implications, so for now I'm happy to use it. My personal experience with ORMs is that they're slow and comebersome, but that might be just a PHP land issue.

### PR DESCRIPTION
**Pull request by Maelstromeous**
Implemented Mongoose and MetagameEvent handling including writing to database and updating on end.

Closes #48

### CODE REVIEW

File: app/src/services/mongo/index.ts [hunk 1]
```diff
@@ -18,4 +22,20 @@ export default new ContainerModule((bind) => {
...
+        .inSingletonScope();
+
+    bind<MongooseModelFactory<AlertDeathInterface>>('AlertDeathModelFactory')
+        .toDynamicValue(({container}: Context) => new MongooseModelFactory(
+            container.get(Mongoose),
+            'AlertDeath',
+            alertDeathSchema,
+        ))
+        .inSingletonScope();
 });
```
Comment (Maelstromeous) [inline app/src/services/mongo/index.ts:41]: Here is where all of the models will be defined and available to be injected into handlers.

File: app/src/handlers/AlertHandler.ts [hunk 2]
```diff
@@ -12,25 +17,35 @@ interface Alert {
...
+
     private _alerts: Alert[] = [];
 
-    public handleMetagameEvent(mge: MetagameEventEvent): boolean {
+    constructor(
+    @inject('AlertModelFactory')
+        factory: MongooseModelFactory<AlertInterface>,
+    ) {
+        this.factory = factory;
+    }
```
Comment (NotThatBowser) [inline app/src/handlers/AlertHandler.ts:32]: If event handler constructors are only used for injection, we could use the property injection format to remove repetition & allow those properties to self-document more clearly.

For example with this handler, that would mean removing the constructor and changing line 23 to:

``` typescript
@inject('AlertModelFactory')
private readonly factory!: MongooseModelFactory<AlertInterface>;
```

The added **!** on factory is to indicate to the compiler that it won't be null, so it won't warn that it's unassigned.

If we decided to use this format, it would need applying to the other handlers.

Comment (microwavekonijn) [reply app/src/handlers/AlertHandler.ts:32]: I think this was discussed when creating the Eslint rules. Though I liked it the recommendation was not to use it.

Comment (microwavekonijn) [reply app/src/handlers/AlertHandler.ts:32]: Never mind, I see what you did. Though still thinks that doesn’t work because of the way inversify is configured. Plus with this you are able to generate these classes more easily I would say.

Comment (Maelstromeous) [reply app/src/handlers/AlertHandler.ts:32]: @NotThatBowser while you're right, it is cleaner to do injections via this method, I'm enforcing a project wide ideom of we inject things into constructors from the container and assign it there. Simply for the reason of for newer programmers they will understand better where things are coming from. I don't like too much "magic" going on when I write things, which makes them super hard to follow if someone doesn't fully understand the concept.

File: app/src/handlers/AlertHandler.ts [hunk 3]
```diff
@@ -12,25 +18,34 @@ interface Alert {
...
+
     private static readonly logger = getLogger('AlertHandler');
 
+    private readonly factory: MongooseModelFactory<AlertInterface>;
+
     private _alerts: Alert[] = [];
 
-    public handleMetagameEvent(mge: MetagameEventEvent): boolean {
+    constructor(
+    @inject(TYPES.alertModelFactory) factory: MongooseModelFactory<AlertInterface>,
```
Comment (microwavekonijn) [inline app/src/handlers/AlertHandler.ts:29]: Indentation

File: app/src/handlers/census/DeathEventHandler.ts [hunk 4]
```diff
@@ -7,18 +7,27 @@ import DeathEvent from './events/DeathEvent';
...
 export default class DeathEventHandler implements EventHandlerInterface {
     private static readonly logger = getLogger('DeathEventHandler');
 
     private readonly playerHandler: PlayerHandlerInterface;
 
-    constructor(@inject(TYPES.playerHandlerInterface) playerHandler: PlayerHandlerInterface) {
+    private readonly alertDeathModelFactory: MongooseModelFactory<AlertDeathInterface>;
+
+    constructor(
+    @inject(TYPES.playerHandlerInterface) playerHandler: PlayerHandlerInterface,
```
Comment (microwavekonijn) [inline app/src/handlers/census/DeathEventHandler.ts:23]: Indentation

**Comment by Maelstromeous** [comment]
> Looks good. Do you still need the priority feature for services? Or are you already working on that?

I've managed to get away with it without requiring it. Check branch features/45-death-event
